src/librustdoc/html/static/main.js

```diff
@@ -890,6 +890,22 @@
         return "\u2212"; // "\u2212" is '−' minus sign
     }
 
+    $(".search-input").on("keyup",function() {
```

Comment:
if ($(this).val().length === 0) {
    window.history.replaceState("", "std - Rust", "?search=");
    $('#main.content').removeClass('hidden');
    $('#search.content').addClass('hidden');
} 

??